### DISCUSSION TITLE
Adding original folder keywords as label (Like import)

### DISCUSSION THREAD

**Discussion by goyban** [photoprism/photoprism#4335] [Q&A] [answered]
What works:
Original file and folder names as keywords when importing. As mentioned in docs:

> The original file and folder names are used to extract keywords. For example, when you index a folder with the path "Vacation/Africa", all files from this folder will get the keywords "vacation" and "africa" ([link](https://docs.photoprism.app/user-guide/library/#importing-files)).

When I Import new photos path and folder names will be added as keywords which makes it much easier to find the photos later since I have some ideas what I've named them before.

What doesn't work (My question):
Original file and folder names as keywords being extracted from Originals.

So I don't know if I'm missing it or what, but I have a large originals archive with the same method of naming the folders and sub folders and I would like to be able to find the photos in Photoprism using those labels as well. Is there a way to make Photoprism to perform keyword extracting on them as well?

**Comment by graciousgrey** [comment, marked as answer]
Creating keywords from file and folder names should also work when indexing originals. I just checked our own instances and it seems to work as expected. 
Can you give some examples of folder names that are not added as keywords in your case? There a list of stop words that are not used as keywords, numbers will also not be used at this time.

**Reply by goyban** [reply]
Thank you so much for the answer.
I see now, so basically one the main naming methods for me is the source of the photo, for example "Nokia, Telegram, iPad, ..." and this is the main thing I use to search the photos that's why I thought its not working. I can see most folder names are already there as keywords. I could search all these words back when I used to import everything.

I found the stop words and it contains most words I use for naming folders.
Can I manually change stop words since the first thing I remember about a photo is its source (application name or camera/phone name).

**Reply by lastzero** [reply]
It's compiled in. We could load the list dynamically in a future release.

**Reply by goyban** [reply]
Thanks, for now I'll try to change my naming method, but I look forward to dynamic load release!
